website/server/models/challenge.js

```diff
@@ -234,8 +234,21 @@ async function _addTaskFn (challenge, tasks, memberId) {
     }));
   });
 
-  // Update the user
-  toSave.unshift(User.update({ _id: memberId }, updateTasksOrderQ).exec());
+  // Update the tag list of the user document of a participating member of the challenge
+  // such that a tag representing the challenge into which the task to be added will
+  // be added to the user tag list if and only if the tag does not exist already.
+  const addToChallengeTagSet = {
+    $addToSet: {
+      tags: {
+        id: challenge._id,
+        name: challenge.name,
```

Comment:
I just tested this and while the code works correctly it creates a new tag when you already have the original one because `tag.name` should be `challenge.shortName` not `challenge.name`, see https://github.com/HabitRPG/habitica/blob/b947c714f0efb6de1b0558fff885913168890c82/website/server/models/challenge.js#L142